src/main/java/blusunrize/immersiveengineering/client/gui/elements/GuiButtonState.java

```diff
@@ -14,21 +14,52 @@
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.FontRenderer;
 
-public class GuiButtonState extends GuiButtonIE
+public class GuiButtonState<E extends Enum> extends GuiButtonIE
```

Comment:
Are you actually using the `Enum` type bound for anything? If not you could just remove it as well as the special casing for `boolean`s.

Reply:
Uh, yeah, `GuiButtonDyeColor` uses it, and so does the button for the batching state in in `ItemBatcherScreen`
The whole point was to make creating those kinds of buttons easier ^^

Reply:
That's not what I mean. Is there any code that would break if I were to change `GuiButtonState<E extends Enum>` by `GuiButtonState<E>` and then create an instance where `E` is e.g. `String`? I haven't seen any, but I may have missed it.

Reply:
Hmm, no, I don't think there is?

Reply:
You're thinking that would be be a better way to handle the booleans, and also allow other things to by "cycled" in the future, without relying on enums?

Reply:
Yes.